controllers/packagebundlecontroller_controller_test.go

```diff
@@ -67,162 +66,130 @@ func TestPackageBundleControllerReconcilerReconcile(t *testing.T) {
 		}
 	}
 
-	t.Run("marks status inactive if name doesn't match", func(t *testing.T) {
+	t.Run("happy path", func(t *testing.T) {
 		t.Parallel()
 
 		ctx := context.Background()
 		mockClient := mocks.NewMockClient(gomock.NewController(t))
 		pbc := givenPackageBundleController()
-		pbc.Namespace = "blah"
-		inactiveController := types.NamespacedName{
-			Name:      "blah",
-			Namespace: api.PackageNamespace,
-		}
-		mockClient.EXPECT().Get(ctx, inactiveController, gomock.Any()).
+
+		mockClient.EXPECT().Get(ctx, req.NamespacedName, gomock.Any()).
 			DoAndReturn(setMockPBC(&pbc))
-		mockStatusClient := mocks.NewMockStatusWriter(gomock.NewController(t))
-		mockClient.EXPECT().Status().Return(mockStatusClient)
-		mockStatusClient.EXPECT().Update(ctx, gomock.Any(), gomock.Any()).
-			DoAndReturn(func(ctx context.Context, pbc *api.PackageBundleController,
-				opts *client.UpdateOptions) error {
-				if pbc.Status.State != api.BundleControllerStateIgnored {
-					t.Errorf("expected state to be set to Ignored, got %q",
-						pbc.Status.State)
-				}
-				return nil
-			})
+		mockBundleManager := bundleMocks.NewMockManager(gomock.NewController(t))
+		mockBundleManager.EXPECT().ProcessBundleController(ctx, &pbc).Return(nil)
 
-		r := NewPackageBundleControllerReconciler(mockClient, nil, bm,
+		r := NewPackageBundleControllerReconciler(mockClient, nil, mockBundleManager,
 			logr.Discard())
-		result, err := r.Reconcile(ctx, ctrl.Request{NamespacedName: inactiveController})
+		result, err := r.Reconcile(ctx, req)
```

Comment:
If you want to use testify (and I'm for that), you could replace this with:
```
assert.NoError(t, err)
```
That way this function uses testify all the way through, instead of half-and-half. You're probably aware, but the functions in testify's `assert` package have the same semantics as `t.Errorf`, as in they continue the execution of the test. The functions in testify's `require` package correlate to `t.Fatalf`.

Reply:
I strongly agree. In another PR though.